Zend/tests/readonly_props/readonly_clone.phpt

```diff
@@ -0,0 +1,63 @@
+--TEST--
+clone can write to readonly properties
+--FILE--
+<?php
+
+class Counter
+{
+    private static int $counter = 0;
+
+    public readonly int $count;
+
+    public function __construct()
+    {
+        $this->count = ++self::$counter;
+    }
+
+    public function count(?int $count = null): static
+    {
+        $new = clone $this;
+        $new->count = $count ?? ++self::$counter;
```

Comment:
When do you consider that a clone cannot be modified anymore then ?

Can I do something like
```
$new = clone $this;
$this->someService->doThings($new);
$new->count = $count ?? ++self::$counter;
```
?

Wouldn't be better (if possible) to find a new syntax like

1)
```
$new = clone $this {
    $this->count = $count;
}
```
Or

2)
```
$new = clone($count) $this;
```
with
```
public function __clone(?int $count = null) 
{
    // ...
}
```
And the rule would be that you can only set/unset in the __clone method.

Reply:
> When do you consider that a clone cannot be modified anymore then ?
> Can I do something like [...]

The rules are exactly the as initialization: same scope as declaration scope to write to a readonly property, allowed only once *during* cloning. Unsetting is also allowed during clone, which allows setting in the class declaring the property outside of clone. Same rules as currently really, that's why the patch is small. Give it a try!

> Wouldn't be better (if possible) to find a new syntax like

Adding argument to clone makes no sense to me because this concern (wither-like patterns) should not be exposed as public API.

> $new = clone($count) $this;

Not sure what this would mean. As is, `$count` looks like a variable, it might be confusing.

See https://github.com/php/php-src/pull/9403#discussion_r951690476 for a proposal that could work (to me.)